### ISSUE TITLE
508-defect-2: Search results should have consistent heading structures

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#26335]
# [508-defect-1 :exclamation: Launchblocker](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

## User Story or Problem Statement

- As a screen reader user, I expect to see an `h1` to know what page I am on.
- As a screen reader user, I expect headings to be consistent and logically ordered to match the page's structure

### <del>Missing `h1` on VA.gov search results</del>

<del>The launch blocking portion of this ticket is around VA.gov search results not having a `h1` which can be caught when running an axe scan. Currently, "VA.gov search results" is written as a `h2`. It must be written as a `h1` to resolve this launchblocker.

All pages should have a `h1` because it provides an important navigation point for users of assistive technologies, allowing them to easily find the main content of the page.</del>

### Heading inconsistencies

Heading inconsistencies between the three search results pages can damage habituation for the screen reader experience. Inconsistencies include:
- Find a form has search result # as a `h2`, while <del>VA.gov search results and</del> resources don't
- Resources search results are `h2`s , while <del>VA.gov search results and</del> find a form search results are `spans`
- <del>VA.gov search results page lists "More VA search tools" after the search results as an `h4`, when it should be a `h2` to match how find a form and resources handles headings after the search results</del>

<img width="1792" alt="Screen Shot 2021-06-21 at 12 26 37 PM" src="https://user-images.githubusercontent.com/14154792/122797396-6e703080-d28d-11eb-9307-a57712bc8751.png">
<img width="1792" alt="Screen Shot 2021-06-21 at 12 19 49 PM" src="https://user-images.githubusercontent.com/14154792/122797402-6fa15d80-d28d-11eb-9f4d-a057c2f238b7.png">
<img width="1430" alt="Screen Shot 2021-06-21 at 12 19 13 PM" src="https://user-images.githubusercontent.com/14154792/122797404-70d28a80-d28d-11eb-9c0a-0e8b92a84e98.png">

## Acceptance Criteria

- [x] "VA.gov search results" is a `h1`, not a `h2`
- [x] Headings are logically ordered and match the intended structure of the page (I recommend the designer uses [headingsMap](https://chrome.google.com/webstore/detail/headingsmap/flbjommegcjonpdmenkdiocclhjacmbi) to validate this
- [ ] "Showing X results" is an h2 on resource results
- [ ] Search result titles are headings like in resources and support and VA.gov search results, not spans like in <del>VA.gov search results</del> find a form search results 

## Outstanding Question

Something that should be resolved is the wording of headings on the page when it comes to the _search_ and the _search results_. For example, consider VA.gov search results:

### Before, confusing structure

<img width="1041" alt="Screen Shot 2021-06-22 at 10 55 01 AM" src="https://user-images.githubusercontent.com/14154792/122948522-f7986d80-d348-11eb-8bf9-5d940ae0285b.png">

### After, semantic structure

<img width="1041" alt="Screen Shot 2021-06-22 at 10 58 08 AM" src="https://user-images.githubusercontent.com/14154792/122948544-fbc48b00-d348-11eb-948d-a01af921e20a.png">

## Proposed Solution

I recommend the designer collaborates with the developer and accessibility specialist to create a consistent headings experience for screen reader users. The acceptance criteria above should provide a starting point.

## WCAG or Vendor Guidance 
[Axe Scan (Deque): Page must contain a level-one heading](https://dequeuniversity.com/rules/axe/4.2/page-has-heading-one?application=coconut)

**Comment by mgleo** [comment]
<img width="1104" alt="Screen Shot 2021-06-25 at 9 11 00 AM" src="https://user-images.githubusercontent.com/5553542/123443964-5f96c000-d5a4-11eb-8606-6e5d1dd33b08.png">
@srsuddath Josh shared this screenshot as a solution. The text "More search results" may change depending on conversations with Content team.

**Comment by mgleo** [comment]
@srsuddath Approved by the content team, ok to move forward on

**Comment by joshkimux** [comment]
Confirmed that VA.gov search results has been fixed 🥳 

<img width="1305" alt="Screen Shot 2021-07-06 at 1 22 27 PM" src="https://user-images.githubusercontent.com/14154792/124642456-dca41e00-de5d-11eb-8bbe-bd17968de85e.png">

**Comment by joshkimux** [comment]
@MarciMcGuireGCIO and @zacharymorel forwarding this ticket over your way. Two items remain to get search results to be consistent with one another:

- "Showing X results" should be a h2 on resource results
- Find a form search results should be headings

<img width="1792" alt="Screen Shot 2021-06-21 at 12 26 37 PM" src="https://user-images.githubusercontent.com/14154792/122797396-6e703080-d28d-11eb-9307-a57712bc8751.png">
<img width="1430" alt="Screen Shot 2021-06-21 at 12 19 13 PM" src="https://user-images.githubusercontent.com/14154792/122797404-70d28a80-d28d-11eb-9c0a-0e8b92a84e98.png">

**Comment by MarciMcGuireGCIO** [comment]
@joshkimux forgive my confusion.  I see that the title changed from Defect 1 to Defect 2, but comments still has Defect 1 and there is a Defect 1 flag.  Can you please confirm which this is?  Thanks!

**Comment by joshkimux** [comment]
It's a defect 2, I'll fix that now!

**Comment by MarciMcGuireGCIO** [comment]
Awesome!  Thanks, @joshkimux !

**Comment by laflannery** [comment]
Resources and Support and Find a form search results have proper headings now. Closing issue
